tls: flaky TestHandshakeConnectionCancelations/cancel_outgoing_connection

```
  === RUN   TestHandshakeConnectionCancelations/cancel_outgoing_connection
      transport_test.go:158: 
          	Error Trace:	transport_test.go:158
          	Error:      	An error is expected but got nil.
          	Test:       	TestHandshakeConnectionCancelations/cancel_outgoing_connection
  === RUN   TestHandshakeConnectionCancelations/cancel_incoming_connection
  --- FAIL: TestHandshakeConnectionCancelations (0.04s)
      --- FAIL: TestHandshakeConnectionCancelations/cancel_outgoing_connection (0.01s)
      --- PASS: TestHandshakeConnectionCancelations/cancel_incoming_connection (0.01s)

--- a/p2p/security/tls/transport_test.go
+++ b/p2p/security/tls/transport_test.go
@@ -165,7 +165,13 @@ func TestHandshakeConnectionCancelations(t *testing.T) {
 		go func() {
 			ctx, cancel := context.WithCancel(context.Background())
 			cancel()
-			_, err := serverTransport.SecureInbound(ctx, &delayedConn{Conn: serverInsecureConn, delay: 5 * time.Millisecond}, "")
+			conn, err := serverTransport.SecureInbound(ctx, &delayedConn{Conn: serverInsecureConn, delay: 5 * time.Millisecond}, "")
+			// crypto/tls' context handling works by spinning up a separate Go routine that watches the context,
+			// and closes the underlying connection when that context is canceled.
+			// It is therefore not guaranteed (but very likely) that this happens _during_ the TLS handshake.
+			if err == nil {
+				_, err = conn.Read([]byte{0})
+			}
 			errChan <- err
 		}()
 		_, err = clientTransport.SecureOutbound(context.Background(), clientInsecureConn, serverID)
@@ -188,7 +194,13 @@ func TestPeerIDMismatch(t *testing.T) {
 
 		errChan := make(chan error)
 		go func() {
-			_, err := serverTransport.SecureInbound(context.Background(), serverInsecureConn, "")
+			conn, err := serverTransport.SecureInbound(context.Background(), serverInsecureConn, "")
+			// crypto/tls' context handling works by spinning up a separate Go routine that watches the context,
+			// and closes the underlying connection when that context is canceled.
+			// It is therefore not guaranteed (but very likely) that this happens _during_ the TLS handshake.
+			if err == nil {
+				_, err = conn.Read([]byte{0})
+			}
 			errChan <- err
 		}()
 

